bubblesub/ui/main_window.py

```diff
@@ -208,6 +218,30 @@ def _setup_menu(self) -> None:
             HotkeyContext.Global,
         )
 
+    def _restore_widgets_visibility(self) -> None:
+        for (key, data) in self._api.cfg.opt["gui"]["visibility"].items():
+            widget = self.findChild(QtWidgets.QWidget, key)
+
+            if data[1]:
+                widget = widget.parentWidget()
+
+            widget.setEnabled(data[0])
+            widget.setVisible(data[0])
+
+    def _store_widgets_visibility(self) -> None:
+        def _store_widget(widget_name: str, childs: bool):
+            widget = self.findChild(QtWidgets.QWidget, widget_name)
+
+            if childs:
+                widget = widget.parentWidget()
+
+            return [widget.isVisible(), childs]
+
+        self._api.cfg.opt["gui"]["visibility"] = {
+            "console": _store_widget("console", True),
+            "note-editor": _store_widget("note-editor", False),
+        }
```

Comment:
I'd move these two utility functions to ViewManager TBH as MainWindow is bloated